sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1,12 +1,22 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
+#include "azure/core/platform.hpp"
```

Comment:
Out of curiosity, what required this reordering of the platform.hpp header (along with `WIN32_LEAN_AND_MEAN`)?

If it's required, it would be useful to add that as a comment for future maintainers, and to avoid reordering it again.

Reply:
The @!$@#$ windows headers #define min(x,y). 

And apparently something brings them in before the old code.

This wasn't a problem because nobody used std::min before, but this PR adds in a call to std::min. And thus chaos ensued. This fixes that problem.